content/2015.07.whistler-use-cases-en.rst

```diff
@@ -0,0 +1,171 @@
+
+We can't say too much right now, but the aforementioned shared collections feature in *Kinto* would fit so very nicely into the future of the Firefox browser. :)
+
+Coming Soon
+-----------
+
+In all likelihood we will need to implement indexing and full-text searching (read: Elasticsearch) before the end of the year. This fits nicely into our roadmap since it's something that we had in Daybed already.
+
+.. image:: {filename}/images/whistler-labs.jpg
+    :alt: Firefox Labs Meeting - CC0
+    :align: center
+
+Browser.html
+============
+
+The Search team has been exploring and extending the very notion of a platform, and is currently working on implementing a web browser in Javascript and HTML5 with `React <https://facebook.github.io/react/>`_, the result of which is affectionatly known as `browser.html <https://github.com/mozilla/browser.html>`_.
+
+Kinto aligns very well with the needs of that project as regards the synchronisation of user data. This could be as simple as a replication of Sync itself, but it could also be something more interesting, such as entire collections of any arbitrary data - for example, browser preferences, or an Alexa-top-500-style mechanism that would allow URL completion without the need to send a search request.
+
+But why stop with just the local browser? We could synchronise entire React states between peripherals, thus allowing a seamless browser experience across every device!
+
+Coming Soon
+-----------
+
+If Browser.html must store navigation data, we believe that it should do so using strong encryption directly in the Javascript client. Luckily we're not the only ones thinking about this, and there are already `several possible standards <http://www.w3.org/TR/WebCryptoAPI/>`_.
+
+To avoid pinging the server at regular intervals in order to synchronise the changes (effectively DDoS'ing ourselves!), the introduction of `push notifications <https://w3c.github.io/push-api/>`_ seems like a natural choice. This would be the final building block in our quest to build a complete "Mobile / Web backed as a service".
+
+.. image:: {filename}/images/whistler-roadmap.jpg
+    :alt: Roadmap - CC0
+    :align: center
+
+Conclusion
+==========
+
+We're in pretty much the ideal situation right now: `everything <https://github.com/mozilla-services/kinto/wiki/Roadmap>`_ that we've imagined, worked on, prototyped, and shipped corresponds to the needs and desires of a number of teams at Mozilla.
+
+Our challenges ahead are:
+
+* Co-ordinate with the other teams efficiently.
+* Avoid `disappointment <https://i.imgur.com/jeFYGQ5.jpg>`_.
+* Maintain a high level of productivity.
+* Contunue to improve and promote our solutions.
+* Focus on quick wins that move us forward.
+
+Finally, we encourage meaningful community contributions to help us build a free, generic, simple, and self-hostable solution for storing data on the web.
```

Comment:
In the original version, what we meant is to bring people on board. I mean that's our responsibility to do everything so that people feel like to contribute.